sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -665,6 +665,9 @@ private[hive] class HiveClientImpl(
     val catalogTable = getTable(db, table)
     val hiveTable = toHiveTable(catalogTable, Some(userName))
     specs.zip(newSpecs).foreach { case (oldSpec, newSpec) =>
+      if (client.getPartition(hiveTable, newSpec.asJava, false) != null) {
```

Comment:
@MaxGekk, are there more places like to fix in this file? I think we can fix in one go instead of fixing one by one.

Reply:
Maybe, there are more places here but I am not fixing them somehow randomly. I find those issues while unifying tests for DSv2, DSv1 In-Memory and Hive Table (SPARK-33381). More tests we will unify, more behavior differences we will find, I guess.

Reply:
In particular, this fix is needed for https://github.com/apache/spark/pull/30863#discussion_r546630850

Reply:
👌